docs/docs/introduction/container-images.md

```diff
@@ -0,0 +1,204 @@
+At the end we did get a single cohesive thing, but it had these very two different cgroups and namespaces.  
+If you are interested in learning about platform specific runtime differences, see the [runtime-spec](https://github.com/opencontainers/runtime-spec/blob/master/implementations.md) repository.
+
+## Containers
+
+Everything that claims to be a container technology is basically orchestrating this cgroup and namespace system. 
+You can draw boxes on your powerpoint, call containers "packages" or "encapsulated environments" or what not - a container technology is just some software to stick it's boogery fingers into these system APIs. Containers: 
+
+ - are encapsulations of an environment
+ - have their own process namespace
+ - the main container on the host is still associated with one process, and when you start that process the set within the container might kick off, and when you kill it the container's processes are also killed.
+
+
+### Container Namespaces
+
+Akin to the namespaces on your host, a container also has its own namespace? 
+Here it is helpful to think of a container like a little packaged environment, one that you can actually shell into like you would a server. 
+When I shell inside the container I find that it has its own little process namespace.
+
+### Container Cgroups
+
+A cgroup is a "control group" and it's exactly that - a feature of the kernel that lets you allocate permissions for resources. 
+So, for example, I might allocate a particular amount of memory, or CPU time, or network bandwidth to a set of tasks, these processes (point to processes) that are running on my computer. 
+What else can we do? 
+We can monitor them, reconfigure them, and even deny them access to resources.
+So these containers also have their own set of cgroups.
+
+## What about a container image?
+
+The words "container" and "image" are sometimes used synonymously, but are subtly two different things. 
+You can think of a container image as a template, and a container is one or more running instances of that template. 
+In more concrete terms, the image is some binary sitting on your filesystem that is ready to instantiate to create an isolated, encapsulated environment.
+
+## Example Interactions
+
+Let's give a concrete example of how you might use a container. 
+Let's say you are a scientist, and a life changing paper just came out that describes how to estimate an optimal cheese eating time based on measured properties of the cheese. 
+You've been collecting cheeses for decades, you can easily collect the metrics needed, and you just need the software. 
+
+### 1. You find container images in a registry
+{: .no-toc }
+
+Thankfully, the publishes also released a container in a registry (such as [Docker Hub](https://hub.docker.com/)) and you can:
+
+ 1. Install the Docker daemon to interact with Docker containers
+ 2. Pull the container to your local machine
```

Comment:
We always encourage users to pull first, because if they launch a job across multiple nodes they all won't find it in the cache, and some will be racing to pull it. That's actually okay because there is atomicity, but at least for Singularity, since a run on the fly, this means we are pulling to a temporary spot and likely throwing the container away after use. If the scientist needed to go back and reproduce the result, it wouldn't be totally clear what container was used ("latest?" right? :P) This is why we have them pull, and pull a specific version, save on $SCRATCH, and then launch the job with the container as a direct path to the file. We also save all the nodes trying to pull at the same time if something isn't cached. Anyway, this is why pull comes before run is in my brain, but you're right is not something I do with Docker.